templates/boilerplate/server-py-aiohttp/services.py

```diff
@@ -21,46 +23,156 @@
+        """connects to the server using a simple interface with host and port
+        
+        Arguments:
+            ip {string} -- the host
+            port {number} -- the port
+            service {thrift.client} -- the service client object
+        """
+        transport = self.create_transport(ip, port)
+        protocol = self.create_binary_protocol(transport)
+        return self.create_client_open_connection(transport, protocol, service)
+
+    def connect_multiplexed_interface(self, ip, port, service_name, service):
+        """connects to the server using a multiplexed interface (with several services on the same port)
+        
+        Arguments:
+            ip {string} -- the host
+            port {number} -- the port
+            service_name {string} -- the service client name
+            service {thrift.client} -- the service client object
+        """
+        from thrift.protocol import TMultiplexedProtocol    
+        
+        transport = self.create_transport(ip, port)
+        protocol = self.create_binary_protocol(transport)
+        # update the protocol to a multiplexed protocol (several services are agregated on the same port)
+        multiplexedProtocol = TMultiplexedProtocol.TMultiplexedProtocol(protocol, service_name)
+        return self.create_client_open_connection(transport, multiplexedProtocol, service)
+
+    def create_clients_std(self, ip, *ports):
+        """creates the client objects using standard connection (1 client per port)
+        
+        Arguments:
+            ip {string} -- host
+            ports {numbers} -- ports (1 per service)
+        """
+        port0, port1 = ports if len(ports) == 2 else (ports[0], ports[0]+1)
+
+        self.applicationClient = self.connect_std_buffer_interface(
+            ip, port0, application.Application)
+
+        self.modelerClient = self.connect_std_buffer_interface(
+            ip, port1, modeler.Modeler)
+
+
+    def create_clients_multiplexed(self, ip, port):
+        """creates the client objects using multiplexed connection (several clients per port)
+        
+        Arguments:
+            ip {string} -- host
+            port {number} -- port
+        """
+        self.applicationClient = self.connect_multiplexed_interface(
+            ip, port, 'Application', application.Application)
+        self.modelerClient = self.connect_multiplexed_interface(
+            ip, port, 'Modeler', modeler.Modeler)
+
+    def create_clients(self, config):
+        """creates the client objects
+        
+        Arguments:
+            config {config} -- configuration object
+        """
+        if config.THRIFT_USE_MULTIPLEXED_SERVER == True:
+            self.create_clients_multiplexed(config.CS_S4L_HOSTNAME, config.CS_S4L_PORT_APP)
+        else:
+            self.create_clients_std(config.CS_S4L_HOSTNAME, config.CS_S4L_PORT_APP, config.CS_S4L_PORT_MOD)
+
+    def __init__(self, config):
+        self.create_clients(config)
+
+def connect():
```

Comment:
**TIP**: Both `connect` and `callRPCFunction` can be defined as [static methods](https://stackoverflow.com/questions/735975/static-methods-in-python#735978) of class `xRpcModelerInterface`. This way you avoid the ugly `global`

Reply:
created an additional class that handles the connection management using static methods.